### PR TITLE
Retrofit.Builder.addConverter().

### PR DESCRIPTION
**Pull request by swankjesse**
Seems like it would be convenient.

### CODE REVIEW

File: retrofit/src/main/java/retrofit/Retrofit.java [hunk 1]
```diff
@@ -205,6 +206,21 @@ public Builder baseUrl(BaseUrl baseUrl) {
...
 
+    /** Add converter for serialization and deserialization of {@code type}. */
+    public <T> Builder addConverter(final Type type, final Converter<T> converter) {
+      checkNotNull(type, "type == null");
+      checkNotNull(converter, "converter == null");
+      converterFactories.add(new Converter.Factory() {
+        @Override public Converter<?> get(Type candidate) {
+          return candidate.equals(type) ? converter : null;
+        }
+        @Override public String toString() {
```
Comment (swankjesse) [inline retrofit/src/main/java/retrofit/Retrofit.java]: (We don't do a great job of overriding `toString()` throughout Retrofit)

File: retrofit/src/main/java/retrofit/http/Part.java [hunk 2]
```diff
@@ -18,6 +18,7 @@
 import java.lang.annotation.Documented;
 import java.lang.annotation.Retention;
 import java.lang.annotation.Target;
+import retrofit.Converter;
 import retrofit.Retrofit;
```
Comment (JakeWharton) [inline retrofit/src/main/java/retrofit/http/Part.java]: Unused import

File: retrofit/src/main/java/retrofit/http/PartMap.java [hunk 3]
```diff
@@ -18,6 +18,7 @@
 import java.lang.annotation.Documented;
 import java.lang.annotation.Retention;
 import java.lang.annotation.Target;
+import retrofit.Converter;
 import retrofit.Retrofit;
```
Comment (JakeWharton) [inline retrofit/src/main/java/retrofit/http/PartMap.java]: Unused import

**Comment by JakeWharton** [comment]
LGTM

**Pull request merged**